lib/ecto/repo.ex

```diff
@@ -428,6 +432,32 @@ defmodule Ecto.Repo do
   @callback all(queryable :: Ecto.Query.t, opts :: Keyword.t) :: [Ecto.Schema.t] | no_return
 
   @doc """
+  Returns a lazy enumerable that emits all entries from the data store
+  matching the given query.
+
+  May raise `Ecto.QueryError` if query validation fails.
+
+  ## Options
+
+    * `:prefix` - The prefix to run the query on (such as the schema path
+      in Postgres or the database in MySQL). This overrides the prefix set
+      in the query.
+
+  See the "Shared options" section at the module documentation.
+
+  ## Example
+
+      # Fetch all post titles
+      query = from p in Post,
+           select: p.title
+      stream = MyRepo.stream(query)
+      MyRepo.transaction(fn() ->
```

Comment:
It is required for SQL adapters but others may not required it. I will add that to the docs though. We require it inside a transaction so we run the same code and get the same behaviour in and outside the sandbox. If we didn't use transaction it may be hard to understand when a connection is in use. It also means we can keep pooling completely separate from streaming.